### DISCUSSION TITLE
Migration guide for 1.3 -> 1.4

### DISCUSSION THREAD

**Discussion by rahulahoop** [mapstruct/mapstruct#2752] [General]
Hi there

Just wondering if it would be possible to develop a clear cut migration guide. we are using 1.3.2 that is deeply integrated in our project, filled with inherited configs, decoraters, and polymorphism. When attempting to upgrade to 1.4 we get a load of erros and its a bit intimidating to know what will be affected purely by the breaking changes lists.

Thanks in advance !

**Comment by Zegveld** [comment]
I don't know if I'll have time to work on this though, but here is the list of release/migration notes for creating such a migration guide:

Release/Migration notes for versions 1.4.0 through 1.4.2:
* https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.Final
  * https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.Beta1
  * https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.Beta2
  * https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.Beta3
  * https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.CR1
* https://github.com/mapstruct/mapstruct/releases/tag/1.4.1.Final
* https://github.com/mapstruct/mapstruct/releases/tag/1.4.2.Final

**Reply by filiphr** [reply]
Most of the changes are part of the [1.4.0.Beta1](https://github.com/mapstruct/mapstruct/releases/tag/1.4.0.Beta1) release notes. I don't think that it is worth it in putting everything into an additional guide, since anyone can easily go through the Beta release notes.

What could be done for the future is to include the release notes of the Beta and CR releases into the Final release as well.
